okhttp/src/main/java/io/grpc/okhttp/OkHttpTlsUpgrader.java

```diff
@@ -33,6 +33,7 @@
 
 import com.google.common.base.Preconditions;
 
+import com.squareup.okhttp.internal.tls.OkHostnameVerifier;
```

Comment:
(Android 6.0 is loading and using it directly)[https://android.googlesource.com/platform/libcore/+/android-6.0.0_r7/luni/src/main/java/javax/net/ssl/HttpsURLConnection.java#115].

But I don't mind having it in our repository given we've already had some.

Pulled OkHostnameVerifier and the depended DistinguishedNameParser into our repository from okhttp 2.5.0 with a few minor changes that fills the diamond operator.